src/main/java/step3/RaceCondition.java

```diff
@@ -0,0 +1,33 @@
+package step3;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class RaceCondition {
+
+    private final List<Car> cars;
+
+    public RaceCondition(Integer participationCount) {
+        this.cars = this.prepareCars(participationCount);
+    }
+
+    AttemptResult attempt() {
+        return new AttemptResult(orderGo());
+    }
+
+    Integer getCarsCount() {
+        return this.cars.size();
+    }
+
+    List<CarWentResult> orderGo() {
+        return this.cars.stream().map(car -> car.go()).collect(Collectors.toList());
```

Comment:
가독성을 위해 아래처럼 개행해주셔도 좋습니다:)
```suggestion
        return this.cars.stream()
        .map(car -> car.go())
        .collect(Collectors.toList());
```